src/NServiceBus.Core/Transports/IDequeueMessages.cs

```diff
@@ -2,23 +2,21 @@
 {
     using System;
 
-    //public class Dequeue
-
     /// <summary>
     /// Interface to implement when developing custom dequeuing strategies.
     /// </summary>
-    public interface IDequeueMessages : IObservable<MessageAvailable>
+    public interface IDequeueMessages
     {
         /// <summary>
         /// Initializes the <see cref="IDequeueMessages"/>.
         /// </summary>
-        DequeueInfo Init(DequeueSettings settings);
-        
+        DequeueInfo Init(Action<ProcessingContext> pipe, DequeueSettings settings);
+
         /// <summary>
         /// Starts the dequeuing of message/>.
         /// </summary>
-        void Start();
-        
+        void Start(DequeueLimitations limitations);
```

Comment:
Don't like the name `DequeueLimitations` perhaps `RuntimeSettings` or `StartContext` or similar?